src/core/lobpcg_hyper_impl.jl

```diff
@@ -8,14 +8,22 @@ vprintln(args...) = nothing
 
 
 using LinearAlgebra
-# using LazyArrays
+using GenericLinearAlgebra
 
 # Perform a Rayleigh-Ritz for the N first eigenvectors.
 function RR(X, AX, BX, N)
     F = eigen(Hermitian(X'AX))
     F.vectors[:,1:N], F.values[1:N]
 end
 
+import LinearAlgebra: cholesky
+function cholesky(X::Union{Matrix{ComplexF16}, Hermitian{ComplexF16,Matrix{ComplexF16}}})
+    # Cholesky factorisation above may promote the type
+    # (e.g. Float16 is promoted to Float32. This undoes it)
+    U = cholesky(ComplexF32.(X)).U
+    (U=convert.(ComplexF16, U), )
+end
```

Comment:
Should that be a PR to Base?

Reply:
Well ... they've been looking for someone to do it properly. See issue 16446. At the moment I do not have time to do this, also I'm not sure it's worth the effort DFTK in Float16 sounds like a bad idea anyway...